### PR TITLE
make: add googlefonts target

### PR DESCRIPTION
**Pull request by m4rc1e**
This pull requests adds a `googlefonts` target to the makefile. It will create a new set of variable fonts that comply with the GF spec. Personally, I feel this approach is cleaner than GF maintaining a fork.

@rsms many people have requested an updated version of Inter on Google Fonts so I'm hoping this solution works for you,
https://github.com/google/fonts/issues/3429. We may end up calling the family Inter 4 if we cannot all agree that the regressions are not going to cause people to complain.

### CODE REVIEW

**Comment by m4rc1e** [comment]
Friendly ping @rsms

File: Pipfile [hunk 1]
```diff
@@ -10,15 +10,15 @@ ufo2ft = "==2.30.0"
 fontmake = "==3.5.*"
 fontbakery = "==0.10.4"
 skia-pathops = "==0.8.*"
-gftools = "==0.9.*"
+gftools = "*"
```
Comment (rsms) [inline Pipfile:13]: Please pin the version to whatever GF requires.
Accepting "any version" can break CI (it has happened in the past)

File: misc/makezip2.sh [hunk 2]
```diff
@@ -58,7 +58,8 @@ cp build/fonts/var/InterVariable-Italic.ttf   "$ZIPDIR/InterVariable-Italic.ttf"
 cp build/fonts/static/Inter*.woff2            "$ZIPDIR/web/" &
 cp build/fonts/var/InterVariable.woff2        "$ZIPDIR/web/InterVariable.woff2"
 cp build/fonts/var/InterVariable-Italic.woff2 "$ZIPDIR/web/InterVariable-Italic.woff2"
-cp misc/dist/inter.css                         "$ZIPDIR/web/"
+cp misc/dist/inter.css                        "$ZIPDIR/web/"
+cp build/fonts/googlefonts/Inter*.ttf         "$ZIPDIR/googlefonts/"
```
Comment (rsms) [inline misc/makezip2.sh:62]: (Please revert this change)

Was this accidental? This script (makezip2.sh) is used to create the offician distribution, so it shouldn't contain Google Fonts files.

If the intention was to get a zip file of the `googlefonts` make target, I suggest adding a custom `zip ...` invocation to the `googlefonts` make target.

**Review by rsms — CHANGES_REQUESTED** [review]
Thank you for this. I agree that this is a cleaner solution than GF maintaining a fork :•)

File: misc/makezip2.sh (hunk 2 above)
Comment (mon-jai) [reply misc/makezip2.sh:62]: (Not affiliated with the Google Fonts team)

Why are there `makezip.sh` and `makezip2.sh`? Is the former obsolete?

Comment (rsms) [reply misc/makezip2.sh:62]: @mon-jai something like that, I forget. None of them are intended to be used directly, just to reduce the complexity of the makefile.

Comment (mon-jai) [reply misc/makezip2.sh:62]: Maybe delete the obsolete one too?

Comment (m4rc1e) [reply misc/makezip2.sh:62]: Thanks for the review.

This change is intentional. In order for the fonts to be included in Google Fonts, they must reside in a zip file that's attached to the release. The easiest way for me to do this is to bundle the gf edition fonts into the existing zip. If you want them in a separate zip, I can tweak the github action. Either way, whenever you cut a new version, we'll need the googlefonts fonts to be generated and included with the release.

Comment (rsms) [reply misc/makezip2.sh:62]: Got it! Thanks for clarifying.
Github actions is only used for CI. The actual release I do manually, so, we can add a second zip file e.g. "Inter-VERSION-GoogleFonts.zip" which can then be fetched from github releases (scripted or manually.)
I'll add a patch in a separate comment for you.

**Comment by rsms** [comment]
Patch for making a separate `Inter-VERSION-GoogleFonts.zip` file

```diff
--- /Users/rsms/Downloads/a.mk	2024-03-22 11:40:10.000000000 -0700
+++ /Users/rsms/Downloads/b.mk	2024-03-22 11:43:07.000000000 -0700
@@ -424,6 +424,7 @@
 # - step2 runs tests, then makes a zip archive and updates the website (docs/ dir.)
 
 DIST_ZIP = build/release/Inter-${VERSION}.zip
+DIST_ZIP_GF = $(SRCDIR)/build/release/Inter-$(VERSION)-GoogleFonts.zip
 
 dist:
 	@echo "——————————————————————————————————————————————————————————————————"
@@ -443,13 +444,15 @@
 	$(MAKE) -f $(MAKEFILE) -j$(nproc) clean
 	$(MAKE) -f $(MAKEFILE) -j$(nproc) all
 	$(MAKE) -f $(MAKEFILE) -j$(nproc) test
-	$(MAKE) -f $(MAKEFILE) -j$(nproc) dist_zip dist_docs
+	$(MAKE) -f $(MAKEFILE) -j$(nproc) dist_zip dist_zip_gf dist_docs
 	$(MAKE) -f $(MAKEFILE) dist_postflight
 
 dist_zip: | venv
-	@#. $(VENV) ; python misc/tools/patch-version.py misc/dist/inter.css
 	bash misc/makezip2.sh -reveal-in-finder "$(DIST_ZIP)"
 
+dist_zip_gf: | venv
+	cd "$(FONTDIR)/googlefonts" && zip -q -X -r "$(DIST_ZIP_GF)" *.ttf
+
 dist_docs:
 	$(MAKE) -C docs -j$(nproc) dist
 
@@ -468,7 +471,7 @@
 	@echo ""
 	@echo "——————————————————————————————————————————————————————————————————"
 
-.PHONY: dist dist_preflight dist_step1 dist_step2 dist_zip dist_docs dist_postflight
+.PHONY: dist dist_preflight dist_step1 dist_step2 dist_zip dist_zip_gf dist_docs dist_postflight

```

**Comment by m4rc1e** [comment]
I also don't mind deleting `makezip.sh` since it isn't used anywhere in the chain.

**Comment by rsms** [comment]
@m4rc1e Thank you for the update. I'll hold off on merging until I hear back from you.

File: Makefile [hunk 3]
```diff
@@ -233,6 +233,14 @@ var: \
 	$(FONTDIR)/var/InterVariable.ttf \
 	$(FONTDIR)/var/InterVariable-Italic.ttf
 
+googlefonts: var
+	gftools fix-family $(FONTDIR)/var/*.ttf \
```
Comment (rsms) [inline Makefile:237]: One more thing: These commands need to be prefixed with `. $(VENV) ;` else they will fail unless the user first manually modifies their shell's environment.

I.e.

```
gftools fix-family ...
```

becomes

```
. $(VENV) ; gftools fix-family ...
```

Comment (rsms) [reply Makefile:237]: (Sorry I didn't catch this earlier)

**Comment by m4rc1e** [comment]
Hey,

So sorry about the delay.

I went on a short break and Dave was off as well. The answer is in, we're not going to rename the family. I've also added `. $(VENV) ;` and updated gftools. I needed to update gftools because we changed an api.

**Comment by rsms** [comment]
Thank you for all your help @m4rc1e

**Comment by m4rc1e** [comment]
Thank you so much for the help and merge. Hate to be a bother but could you cut another release and include the gf zip? I can then onboard the family to Google Fonts.

**Comment by rsms** [comment]
@m4rc1e unfortunately the change of gftools caused and implicit upgrade of fontTools which in turn has a bug (or is broken or I'm using it wrong; most likely) that messes up the wght values in the VF.

```
$ make -j var
...
. build/venv/bin/activate ; python misc/tools/bake-vf.py build/fonts/var/.Inter-Roman.var.ttf -o build/fonts/var/InterVariable.ttf
unexpected wght 481.0 (expected 500) <fontTools.ttLib.ttFont.TTFont object at 0x101cc0e50> {'opsz': 14.0, 'wght': 481.0}
unexpected wght 562.0 (expected 600) <fontTools.ttLib.ttFont.TTFont object at 0x101cc0e50> {'opsz': 14.0, 'wght': 562.0}
unexpected wght 643.0 (expected 600) <fontTools.ttLib.ttFont.TTFont object at 0x101cc0e50> {'opsz': 14.0, 'wght': 643.0}
unexpected wght 758.0 (expected 800) <fontTools.ttLib.ttFont.TTFont object at 0x101cc0e50> {'opsz': 14.0, 'wght': 758.0}
```

I have almost no free time (between raising a newborn baby, running a start-up and sleeping a little bit in between) so I'm not sure when I'll get around to fixing the fontTools issues.

We can try reverting [the change to Pipfile](https://github.com/rsms/inter/commit/bde6fa1677862051fe7948d9190592e641470272#diff-230078d672f10d17463a8a6265cad825b790885898256a3365be90685caac58d) but I assume you had a good reason for upgrading it. Alt we could try to pin fontTools to a version that doesn't break the VF STAT table.

**Comment by m4rc1e** [comment]
Thank you very much for the update. I have a nine month old who's teething so my sleep is completely wrecked,. I have no problems looking into this issue and opening another PR. I'll keep you posted.
